lib/Sema/CSApply.cpp

```diff
@@ -7527,6 +7527,39 @@ Expr *ConstraintSystem::applySolution(Solution &solution, Expr *expr,
     return nullptr;
   }
 
+  for (auto &e : solution.Conformances) {
+    const auto &conformance = e.second;
+
+    auto anchor = expr;
+    if (auto *locator = e.first)
+      anchor = locator->getAnchor();
+
+    if (conformance.isAbstract())
+      continue;
+
+    auto *concrete = conformance.getConcrete();
+    // If conformance check is not yet complete let's re-check using 'Used'
+    // to require conformance check to happen.
+    if (concrete->getState() != ProtocolConformanceState::Complete) {
+      if (auto conformance = TC.conformsToProtocol(
+              concrete->getType(), concrete->getProtocol(), DC,
+              ConformanceCheckFlags::InExpression |
+                  ConformanceCheckFlags::Used)) {
+        if (conformance->isAbstract())
+          continue;
+
+        concrete = conformance->getConcrete();
+      }
+    }
+
+    if (concrete->isComplete() && concrete->isInvalid()) {
```

Comment:
I think forcing the conformance above is overkill. If it happens to be complete and it's invalid, then we should diagnose; otherwise, we'll diagnose it later when the conformance is fully checked.